container-images/tcib/base/os/tempest/run_tempest.sh

```diff
@@ -58,13 +73,82 @@ if [[ ! -z ${TEMPESTCONF_REMOVE} ]]; then
     done <<< "$TEMPESTCONF_REMOVE"
 fi
 
-TEMPESTCONF_OVERRIDES="$(echo ${TEMPESTCONF_OVERRIDES} | tr '\n' ' ') identity.v3_endpoint_type public"
 
 if [ -n "$CONCURRENCY" ] && [ -z ${TEMPEST_CONCURRENCY} ]; then
     TEMPEST_ARGS+="--concurrency ${CONCURRENCY} "
 fi
 
-pushd $HOMEDIR
+function run_git_tempest {
+    mkdir -p $TEMPEST_EXTERNAL_PLUGIN_DIR
+    pushd $TEMPEST_EXTERNAL_PLUGIN_DIR
+
+    python3 -m venv .venv
+    source ./.venv/bin/activate
+
+    for plugin_index in "${!TEMPEST_EXTERNAL_PLUGIN_GIT_URL[@]}"; do
+        git_url=${TEMPEST_EXTERNAL_PLUGIN_GIT_URL[plugin_index]}
+        change_url=${TEMPEST_EXTERNAL_PLUGIN_CHANGE_URL[plugin_index]}
+        refspec=${TEMPEST_EXTERNAL_PLUGIN_REFSPEC[plugin_index]}
+        plugin_name=$(basename -s .git $git_url)
+
+        git clone $git_url
+
+        if [[ ! -z $change_url ]] && [[ ! -z $refspec ]] || \
+           [[ $change_url != "-" ]] && [[ $refspec != "-" ]]; then
+            pushd $plugin_name
+            git fetch $change_url $refspec
+            git checkout FETCH_HEAD
+            popd
+        fi
+
+        pip install ./$plugin_name
```

Comment:
pip/pip3? does it make a difference? I guess pip points to pip3 so it doesn't matter

Reply:
I think that using pip should be ok here. We are inside a virtual environment.